### ISSUE TITLE
Handle confirm_form validation errors gracefully

### ISSUE THREAD

**Issue by 19greg96** [django-import-export/django-import-export#1901]
**Describe the bug**
In case the `confirm_form` is overriden in the admin (as described [here](https://django-import-export.readthedocs.io/en/latest/admin_integration.html#customize-admin-import-forms)) and the form is invalid, currently an unhelpful exception is raised. Example:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.10/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/utils/decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
  File "/usr/local/lib/python3.10/site-packages/django/utils/decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/django/views/decorators/cache.py", line 81, in _view_wrapper
    add_never_cache_headers(response)
  File "/usr/local/lib/python3.10/site-packages/django/utils/cache.py", line 293, in add_never_cache_headers
    patch_response_headers(response, cache_timeout=-1)
  File "/usr/local/lib/python3.10/site-packages/django/utils/cache.py", line 284, in patch_response_headers
    if not response.has_header("Expires"):

Exception Type: AttributeError at /main/somemodel/process_import/
Exception Value: 'NoneType' object has no attribute 'has_header'
```

**To Reproduce**
1. Follow the [Customize admin import forms](https://django-import-export.readthedocs.io/en/latest/admin_integration.html#customize-admin-import-forms) tutorial
2. Add a field  to the confirmation form that can be filled with invalid data by the user
3. Submit the form with invalid data
4. Exception occurs

**Expected behavior**
The proper way of hangling this would be to redirect the user back to the confirm form, with the apropriate validation errors displayed.

**Additional context**
The root cause is that the `confirm_form.is_valid() == False` case is not handled, in this path, `None` is implicitly returned in the view method.

https://github.com/django-import-export/django-import-export/blob/main/import_export/admin.py#L157-L178

**Comment by matthewhegarty** [comment]
Thanks for raising - do you think you could raise a PR to address this?

**Comment by 19greg96** [comment]
Related: https://github.com/django-import-export/django-import-export/pull/1343

> Thanks for raising - do you think you could raise a PR to address this?

Hi Matt, I did not offer raising a PR in the op because I do not know when I can get around to it, but if this is not a high priority, I can take a look (maybe sometime next month?).

**Comment by matthewhegarty** [comment]
No problem, it is not a high priority.  If I get to it first, I'll assign the issue to myself.

**Comment by AyushDharDubey** [comment]
I am unable to reproduce the error at my end.

![image](https://github.com/django-import-export/django-import-export/assets/142207449/f87c3733-88c8-454a-94ef-7a51dd5658fd)

It seems like the error is being displayed as expected. Am I missing something?

**Comment by 19greg96** [comment]
> I am unable to reproduce the "unhelpful exception" at my end.
> 
> ![image](https://private-user-images.githubusercontent.com/142207449/347553024-f87c3733-88c8-454a-94ef-7a51dd5658fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2NDA3NzcsIm5iZiI6MTcyMDY0MDQ3NywicGF0aCI6Ii8xNDIyMDc0NDkvMzQ3NTUzMDI0LWY4N2MzNzMzLTg4YzgtNDU0YS05NGVmLTdhNTFkZDU2NThmZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxMFQxOTQxMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYTA3ZjcyZmZkN2JlNDcxOTk1NWE5Y2ExNzQzMWUxNGJhMzEyNWZlYWIxYmRkNzIwOWUxZmVjZWY5OWQ0YjQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aEpI_e-jctFjtq1v7tFnaldeazR0MTcD4_y9PAFX8Pk)
> 
> It seems like the error is being displayed as expected. Am I missing something? PS: I did follow [Customize admin import forms](https://django-import-export.readthedocs.io/en/latest/admin_integration.html#customize-admin-import-forms) tutorial

It's the confirmation form that has to be invalid, not the data itself. I didn't try the tutorial itself, but I suspect if the author selector is left empty, it will trigger the issue (however, this may not work if the browser does not let you submit).

I stumbled upon this when I added a checkbox to the confirm form, and forgot to set the required attribute to False. This causes the form to be invalid when the checkbox is left unchecked. If you can't reproduce it, I can try to create an mvp.

**Comment by 19greg96** [comment]
I clarified the OP that the invalid field has to be added to the confirmation from.

**Comment by AyushDharDubey** [comment]
@19greg96 Thanks for responding.

**Comment by AyushDharDubey** [comment]
@matthewhegarty may I create a PR to fix this issue?


**Solution**
check if confirm_form.is_valid() == False:
return the template "admin/import_export/import.html" with context['errors'] =  confirm_form.errors

**Comment by matthewhegarty** [comment]
Yes, feel free to create a PR.  Please ensure you follow the [contribution guidelines](https://django-import-export.readthedocs.io/en/latest/contributing.html).

**Comment by matthewhegarty** [comment]
I can reproduce this in the test system by updating `core.forms.CustomConfirmImportForm`:

```python
class CustomConfirmImportForm(AuthorFormMixin, ConfirmImportForm):
    """Customized ConfirmImportForm, with author field required"""

    def clean_author(self):
        if self.data["author"] == "5":
            raise ValidationError("invalid author")
```

I have confirmed with manual testing that the fix in #1905 works.
